metaparser/androidartifact/apk_info.go

```diff
@@ -0,0 +1,132 @@
+package androidartifact
+
+import (
+	"bytes"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/avast/apkparser"
+	"github.com/bitrise-io/go-android/v2/sdk"
+	"github.com/bitrise-io/go-utils/command"
+)
+
+func GetAPKInfoWithFallback(logger Logger, apkPth string) (Info, error) {
+	parsedInfo, err := GetAPKInfo(apkPth)
+	if err == nil {
+		return parsedInfo, nil
+	}
+	// err != nil
+	logger.Warnf("Failed to parse APK info: %s", err)
+	logger.APKParseWarnf("apk-parse", "apkparser package failed to parse APK, error: %s", err)
+
+	return GetAPKInfoWithAapt(apkPth)
+}
+
+type manifest struct {
+	XMLName     xml.Name `xml:"manifest"`
+	VersionCode string   `xml:"versionCode,attr"`
+	VersionName string   `xml:"versionName,attr"`
+	PackageName string   `xml:"package,attr"`
+	Application application
+	UsesSdk     usesSdk
+}
+
+type application struct {
+	XMLName     xml.Name `xml:"application"`
+	PackageName string   `xml:"name,attr"`
+	AppName     string   `xml:"label,attr"`
+}
+
+type usesSdk struct {
+	XMLName       xml.Name `xml:"uses-sdk"`
+	MinSDKVersion string   `xml:"minSdkVersion,attr"`
+}
+
+// GetAPKInfo returns infos about the APK.
+func GetAPKInfo(apkPath string) (Info, error) {
+	var manifestContent bytes.Buffer
+	enc := xml.NewEncoder(&manifestContent)
+	enc.Indent("", "\t")
+
+	zipErr, resErr, manErr := apkparser.ParseApk(apkPath, enc)
+	if zipErr != nil {
+		return Info{}, fmt.Errorf("failed to unzip the APK, error: %s", zipErr)
+	}
+	if resErr != nil {
+		return Info{}, fmt.Errorf("failed to parse resources, error: %s", zipErr)
+	}
+	if manErr != nil {
+		return Info{}, fmt.Errorf("failed to parse AndroidManifest.xml, error: %s", zipErr)
+	}
+
+	var manifest manifest
+	if err := xml.Unmarshal(manifestContent.Bytes(), &manifest); err != nil {
+		return Info{}, fmt.Errorf("failed to unmarshal AndroidManifest.xml, error: %s", err)
+	}
+
+	return Info{
+		AppName:           manifest.Application.AppName,
+		PackageName:       manifest.PackageName,
+		VersionCode:       manifest.VersionCode,
+		VersionName:       manifest.VersionName,
+		MinSDKVersion:     manifest.UsesSdk.MinSDKVersion,
+		RawPackageContent: manifestContent.String(),
+	}, nil
+}
+
+func GetAPKInfoWithAapt(apkPth string) (Info, error) {
+	androidHome := os.Getenv("ANDROID_HOME")
+	if androidHome == "" {
+		return Info{}, errors.New("ANDROID_HOME environment not set")
+	}
+
+	sdkModel, err := sdk.New(androidHome)
```

Comment:
`sdk.NewDefaultModel` would be better because it handles both `ANDROID_HOME` and `ANDROID_SDK_ROOT` env vars.